mcumgr: lib: cmd: img_mgmt: Warning about struct visibility emitted with certain Kconfig options

**Describe the bug**
Observed when building the `smp_svr` sample application with a mix of Kconfig options:
```
In file included from zephyr/samples/subsys/mgmt/mcumgr/smp_svr/src/main.c:22:
zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/img_mgmt.h:246:31: warning: 'struct mgmt_ctxt' declared inside parameter list will not be visible outside of this definition or declaration
  246 | int img_mgmt_error_rsp(struct mgmt_ctxt *ctxt, int rc, const char *rsn);
      |                               ^~~~~~~~~
[125/233] Building C object zephyr/subsys/mgmt/mcumgr/CMakeFiles/subsys__mgmt__mcumgr.dir/lib/cmd/img_mgmt/src/img_mgmt_util.c.obj
In file included from zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt_util.c:14:
zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/img_mgmt.h:246:31: warning: 'struct mgmt_ctxt' declared inside parameter list will not be visible outside of this definition or declaration
  246 | int img_mgmt_error_rsp(struct mgmt_ctxt *ctxt, int rc, const char *rsn);
      |                               ^~~~~~~~~
[139/233] Building C object zephyr/subsys/mgmt/mcumgr/CMakeFiles/...__mgmt__mcumgr.dir/lib/cmd/img_mgmt/src/zephyr_img_mgmt_log.c.obj
In file included from zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/zephyr_img_mgmt_log.c:8:
zephyr/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/img_mgmt.h:246:31: warning: 'struct mgmt_ctxt' declared inside parameter list will not be visible outside of this definition or declaration
  246 | int img_mgmt_error_rsp(struct mgmt_ctxt *ctxt, int rc, const char *rsn);
```

**Expected behavior**
No build warnings

**Impact**
Annoyance

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.14.0
 - Commit SHA or Version used: PR https://github.com/zephyrproject-rtos/zephyr/pull/43878 but unrelated to PR

--- a/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/img_mgmt.h
+++ b/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/include/img_mgmt/img_mgmt.h
@@ -9,6 +9,7 @@
 
 #include <inttypes.h>
 #include "img_mgmt_config.h"
+#include "mgmt/mgmt.h"
 #include <zcbor_common.h>
 
 struct image_version;

--- a/subsys/mgmt/mcumgr/lib/mgmt/include/mgmt/mgmt.h
+++ b/subsys/mgmt/mcumgr/lib/mgmt/include/mgmt/mgmt.h
@@ -8,6 +8,7 @@
 #define H_MGMT_MGMT_
 
 #include <inttypes.h>
+#include "mgmt/mcumgr/buf.h"
 
 #ifdef __cplusplus
 extern "C" {

